src/home/spaces_dock.rs

```diff
@@ -102,51 +138,93 @@ live_design! {
         }
     }
 
+    SDock = {{SDock}} {}
+
     SpacesDock = <AdaptiveView> {
-        Desktop = {
-            flow: Down, spacing: 15
-            align: {x: 0.5}
-            padding: {top: 40., bottom: 20.}
-            width: 68., height: Fill
-            show_bg: true
-            draw_bg: {
-                color: (COLOR_SECONDARY)
+            Desktop =  <SDock> {
+                    flow: Down, spacing: 15
+                    align: {x: 0.5}
+                    padding: {top: 40., bottom: 20.}
+                    width: 68., height: Fill
+                    show_bg: true
+                    draw_bg: {
+                        color: (COLOR_SECONDARY)
+                    }
+
+                    profile = <Profile> {}
+
+                    <Separator> {}
+
+                    <Home> {}
+
+                    <Filler> {}
+
+                    <Settings> {}
             }
+            Mobile = <SDock> {
+                    flow: Right
+                    align: {x: 0.5, y: 0.5}
+                    padding: {top: 10, right: 10, bottom: 10, left: 10}
+                    width: Fill, height: Fit
+                    show_bg: true
+                    draw_bg: {
+                        color: (COLOR_SECONDARY)
+                    }
 
-            <Profile> {}
+                    <Filler> {}
 
-            <Separator> {}
+                    profile = <Profile> {}
 
-            <Home> {}
-            
-            <Filler> {}
+                    <Filler> {}
 
-            <Settings> {}
-        }
+                    <Home> {}
 
-        Mobile = {
-            flow: Right
-            align: {x: 0.5, y: 0.5}
-            padding: {top: 10, right: 10, bottom: 10, left: 10}
-            width: Fill, height: Fit
-            show_bg: true
-            draw_bg: {
-                color: (COLOR_SECONDARY)
+                    <Filler> {}
+
+                    <Settings> {}
+
+                    <Filler> {}
             }
+    }
+}
+
+#[derive(Live, LiveHook, Widget)]
+pub struct SDock {
+    #[deref]
+    view: View,
+    #[rust(None)]
+    verification_state: Option<matrix_sdk::encryption::VerificationState>,
```

Comment:
The `Option` wrapper is unnecessary here -- we always start out in the `Unknown` state. 

When applying this suggestion, also be sure to import `matrix_sdk::encryption::VerificationState`.

```suggestion
    #[rust(VerificationState::Unknown)] verification_state: VerificationState,
```